cpp/oneapi/dal/algo/svm/backend/cpu/train_kernel.cpp

```diff
@@ -53,12 +53,45 @@ static train_result call_daal_kernel(const context_cpu& ctx,
 
     // TODO: data is table, not a homogen_table. Think better about accessor - is it enough to have just a row_accessor?
     auto arr_data    = row_accessor<const Float>{ data }.pull();
-    auto arr_label   = row_accessor<const Float>{ labels }.pull();
     auto arr_weights = row_accessor<const Float>{ weights }.pull();
 
+    auto arr_label     = row_accessor<const Float>{ labels }.pull();
+    auto arr_new_label = array<Float>::empty(row_count * 1);
+
+    arr_label.need_mutable_data();
+    Float value_first_class_label  = arr_label[0];
+    Float value_second_class_label = arr_label[1];
+
+    arr_new_label[0] = Float(-1.0);
+    std::int64_t i   = 1;
+    for (; i < row_count; ++i) {
+        if (arr_label[i] == value_first_class_label) {
+            arr_new_label[i] = Float(-1.0);
+        }
+        else {
+            value_second_class_label = arr_label[i];
+            arr_new_label[i]         = Float(+1.0);
+        }
+    }
+    if (value_first_class_label == value_second_class_label) {
+        throw invalid_argument("Input label data should have more one unique label");
+    }
+
+    for (; i < row_count; ++i) {
```

Comment:
This loop never works

Reply:
Thanks, it is bag, the idea was to find two unique elements, then run and transform in this for.

Reply:
fixed